gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -228,8 +231,18 @@ public GrpcCallContext withEndpointContext(EndpointContext endpointContext) {
         endpointContext);
   }
 
+  /**
+   * Backport of {@link #withTimeoutDuration(java.time.Duration)} using {@link
+   * org.threeten.bp.Duration}
+   */
   @Override
-  public GrpcCallContext withTimeout(@Nullable Duration timeout) {
+  @ObsoleteApi("Use withTimeoutDuration(java.time.Duration) instead")
```

Comment:
Not sure what the standard is with @ObsoleteApi and @Deprecated annotations. Do you think we should have some like `<p>Use {@link withTimeout(java.timeDuration)} instead` in the javadocs instead of the annotation or is having it in the annotation is fine?

Reply:
@lqiu96, @blakeli0 [proposed](https://docs.google.com/document/d/1mcCOVtmQbBVfhMSAa8nevXdDHrJ7XAcUA93TCRRO5qU/edit?resourcekey=0-N729Fx2s2qzk8Np0_vq23w&tab=t.0#heading=h.ar0pjtcixw6q) this annotation as a way of tracking public features that will be deprecated and also making customers aware of both imminent deprecation and existence of new features.

Reply:
Yep, I just wasn't sure if the annotation or javadocs would be the preferred place for users to see the replacement method. Also not sure if it matters at all, but just wondering.

Reply:
Based on the doc you linked, I think this fits into the first scenario, so I think we should add this to the Javadoc as well.

Reply:
Modified javadocs accordingly. Now both `@ObsoleteApi` and the associated javadoc comment will suggest the recommended method